pysmarthashtag/api/authentication.py

```
@@ -390,6 +496,381 @@ async def async_auth_flow(self, request: Request) -> AsyncGenerator[Request, Res
+                self.intl_access_token = result.get("accessToken")
+                self.intl_refresh_token = result.get("refreshToken")
+                self.intl_id_token = result.get("idToken")
+                self.intl_user_id = result.get("userId")
+                expires_in = result.get("expiresIn", 86400)
+
+                expires_at = datetime.datetime.now(datetime.timezone.utc) + datetime.timedelta(seconds=expires_in)
+
+                _LOGGER.info("INTL: Login successful for user %s", self.intl_user_id)
+
+            except (KeyError, ValueError, TypeError) as e:
+                raise SmartAPIError(f"Could not parse INTL login response: {e}")
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)